### PR TITLE
[KYUUBI #6134] Support Flink 1.19

### LINKED ISSUE #6134: [FEATURE] Support Flink 1.19
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Flink 1.19 is about to be released, which Kyuubi Flink engine should support.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by wForget**
# :mag: Description
## Issue References 🔗

This pull request fixes #6134

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: externals/kyuubi-flink-sql-engine/src/test/scala/org/apache/kyuubi/engine/flink/operation/FlinkOperationSuite.scala [hunk 1]
```diff
@@ -993,6 +993,8 @@ abstract class FlinkOperationSuite extends HiveJDBCTestHelper with WithFlinkTest
         statement.getConnection.setSchema("db_a")
         val changedSchema = statement.getConnection.getSchema
         assert(changedSchema == "db_a")
+        // reset database to default
+        statement.getConnection.setSchema("default_database")
```
Comment (wForget) [inline externals/kyuubi-flink-sql-engine/src/test/scala/org/apache/kyuubi/engine/flink/operation/FlinkOperationSuite.scala:997]: Flink 1.19 has the following limitations:
```
Cannot drop a database which is currently in use.
```

**Comment by pan3793** [comment]
The final Flink 1.19 VOTE pass is likely to happen after the Kyuubi 1.9 VOTE starts, as Flink 1.19 already entered the RC2 phase, the API should be stable. To make the Flink 1.19 support be included in Kyuubi 1.9, can we merge the code change first and leave tests until the Flink 1.19 official release available?

**Comment by wForget** [comment]
> The final Flink 1.19 VOTE pass is likely to happen after the Kyuubi 1.9 VOTE starts, as Flink 1.19 already entered the RC2 phase, the API should be stable. To make the Flink 1.19 support be included in Kyuubi 1.9, can we merge the code change first and leave tests until the Flink 1.19 official release available?

SGTM, CI (https://github.com/apache/kyuubi/actions/runs/8198173643) has run successfully, let us remove the CI and profile of flink 1.19 first

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
OK, I merged the code change to the master first, please open another PR for supplying CI and continuous RC testing.

**Pull request closed without merging**